tree.go

```diff
@@ -204,32 +204,49 @@ func New() VerkleNode {
 
 // New creates a new leaf node
 func NewLeafNode(stem []byte, values [][]byte) *LeafNode {
```

Comment:
Some notes here:
- I moved the "value detection" to this method now. 
- I refactored the overall previous code to make it clearer (e.g: the `Generator()` previous calls weren't necessary (not even for `c2`), narrow the scope of some variables, etc).